### ISSUE TITLE
Make sharesheet work in airplane mode

### ISSUE THREAD

**Issue by mikemee** [BranchMetrics/android-branch-deep-linking#386]
1. configure an app to use the default sharing as documented here: https://dev.branch.io/getting-started/branch-universal-object/guide/android/#showsharesheet
1. disable internet for the device (e.g. airplane mode)
1. share in the app to clipboard

Expect: URL is shared to the clipboard

Actual: Nothing is pasted to the clipboard

The same thing seems to happen regardless of the share target. The share gives the appearance of success, but nothing actually happens. (Another clue is to share to an app that prompts for more information, e.g. Slack. It does not prompt, but just quietly returns to the app).

Suggestions: 
 * At the very least, document this
 * Better still, allow for some sort of minimal sharing, e.g. always the same link

**Comment by sojanpr** [comment]
@mikemee Thanks for notifying this. We are actually giving a callback in case of link creation failed. Please see BranchLinkShareListener#onLinkShareResponse. It calls back with an error incase of link creation failure. 
  How about adding a  option to fallback to a long url in case of link creation fails?

**Comment by mikemee** [comment]
Thanks @sojanpr. That's good info. 

It also seems that it works ok with no internet if I call `.setDefaultURL(...)` as part of the configuration of `ShareSheetStyle`.

**Comment by mikemee** [comment]
I created a PR as a suggestion on one way to address this.

**Comment by EvangelosG** [comment]
Hi @mikemee can you try out https://github.com/BranchMetrics/android-branch-deep-linking/tree/sharesheetexcludeinclude -- I added some code last night to handle share sheet airplane mode issues.

**Comment by EvangelosG** [comment]
Hey @mikemee -- I resolved this bug in 2.5.7 -- thanks for reporting!

**Comment by mikemee** [comment]
@EvangelosG Ok thanks. Sorry, this was on my list for today. I'll check it in when 2.5.7 is released and report back. Cheers!

**Comment by EvangelosG** [comment]
It was released yesterday, let me know :)

**Comment by mikemee** [comment]
@EvangelosG Just confirmed that it now appears to work ok, even with airport mode on. 

A followup question now though: If I leave the `.setDefaultURL()` code there, then it uses that instead of generating a "live" link.  I'm not sure what the "best practice" is now...  (Note: your link, https://github.com/BranchMetrics/android-branch-deep-linking/tree/sharesheetexcludeinclude, didn't work for me, "Page Not Found", in case that was supposed to explain).

**Comment by EvangelosG** [comment]
Ah, the link is dead now because I deleted the branch! Here's the change: https://github.com/BranchMetrics/android-branch-deep-linking/pull/384/commits/90b7d541ed272f3112f7c2a1c5fde9df739ba4e3#diff-8c3b4cedcd913474009598bc04ba1c7a -- just take a look at ShareLinkManager if you want to see the logic. As you observed, .setDefaultURL overrides the long link returned in airplane mode, I did not want to change the functionality of .setDefaultURL but I did want to fix your issue. IMO best practice would be to use the BranchError we return to let your user know the link was generated, but that it's a dynamic link (long link) as their connection failed!
